src/test/scala/gcd/GcdTesters2.scala

```diff
@@ -0,0 +1,53 @@
+// See README.md for license details.
+
+package gcd
+
+import chisel3._
+import chisel3.tester._
+import org.scalatest.FreeSpec
+import chisel3.experimental.BundleLiterals._
+
+/**
+  * This is a trivial example of how to run this Specification
+  * From within sbt use:
+  * {{{
+  * testOnly gcd.GcdDecoupledTester
+  * }}}
+  * From a terminal shell use:
+  * {{{
+  * sbt 'testOnly gcd.GcdDecoupledTester'
+  * }}}
+  */
+//scalastyle:off magic.number
+class GcdTesters2 extends FreeSpec with ChiselScalatestTester {
+
+  "Gcd should calculate proper greatest common denominator" in {
+    test(new DecoupledGcd(16)) { dut =>
+      dut.input.initSource()
+      dut.input.setSourceClock(dut.clock)
+      dut.output.initSink()
+      dut.output.setSinkClock(dut.clock)
+
+      val testValues = for { x <- 0 to 10; y <- 0 to 10} yield (x, y)
```

Comment:
I'm not sure introducing more advanced Scala stuff in the template is a good idea. Why not just two for loops? Or hardcode a seq of inputs and outputs?

Reply:
I am with Richard. The template code should be as simple as possible without any fancy Scala style stuff.